### PR TITLE
ViewCollection needs a way to allow ordering to be manipulated

### PR DESCRIPTION
**Pull request by justinvdm**

### CODE REVIEW

**Comment by justinvdm** [comment]
Depends on #412

File: go/base/static/js/src/components/structures.js [hunk 1]
```diff
@@ -107,19 +144,38 @@
...
       return this;
+    },
+
+    rearrange: function() {
+      var keys = Array.prototype.slice.call(arguments);
+
+      if (this.ordered && this.arrangeable) {
+        keys.forEach(function(k, i) { this.arranger(this.get(k), i); }, this);
+        this.sort();
+      }
```
Comment (justinvdm) [inline go/base/static/js/src/components/structures.js:183]: I'm really not sure if this is the best way to re-order a `Lookup`. I've been stuck thinking of better alternatives for a while, but can't think of anything better. I'm definitely up for suggestions on this!

Comment (hodgestar) [inline go/base/static/js/src/components/structures.js:183]: I think this seems fine to me. Would it not be easier to just pass a list to `rearrange` to begin with?

**Comment by hodgestar** [comment]
:+1: other than the comment about maybe passing a list into rearrange directly.

File: go/base/static/js/src/components/structures.js (hunk 1 above)
Comment (justinvdm) [inline go/base/static/js/src/components/structures.js:183]: @hodgestar Ya, I wasn't too sure if I should be passing in a list or just args. It probably would be easier as a list, will do that.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**